### PR TITLE
CAT-1129 Userbricks

### PR DESCRIPTION
**Pull request by sjaindl**
Finishes the implementation of userbricks:
- Refactoring
- UX changes
- Lots of Bugfixing ;)
- Adds/fixes tests

I've also rebased on PR #1713 (first 2 commits), as removal of unpack and keep is relevant for userbricks, too. So, when this PR is merged, PR #1713 can be closed!

### CODE REVIEW

**Comment by sjaindl** [comment]
ok to test

File: catroid/res/layout/dialog_userbrick_name.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  ~ GNU Affero General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Affero General Public License
+  ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+            style="@style/DefaultDialog"
+            android:orientation="vertical"
+            android:paddingBottom="0dp"
+            android:theme="@android:style/Theme.Light" >
```
Comment (cheidenreich) [inline catroid/res/layout/dialog_userbrick_name.xml]: please remove the theme property to match the Pocket Code style

Comment (sjaindl) [inline catroid/res/layout/dialog_userbrick_name.xml]: removed.

**Comment by sjaindl** [comment]
Full testrun down to 76 failures (from 82 in develop):
https://jenkins.catrob.at/view/LeeroyMultiJobs/job/CatroidEmulatorAllTestsSerialNightly/15/

@84n4n4: We further need to increase the timeout for uitest.content.interaction (There are many tests for the new userbrick backpack)..

**Comment by cheidenreich** [comment]
In general this looks good to me, seems very robust.
I am not sure if I tested all the functionality.
I like that the big Methods got split up and everything is more readable now.
I'd like to have at least another person to look over this before merge as there are big changes.
please resolve the conflicts with the current develop branch.
for the tests: are the 76 failures the same as in develop or are there some new included?

**Comment by thmq** [comment]
please rebase and reopen.

**Comment by robertpainsi** [comment]
Out of curiosity, since I can't test it myself and the github docu doesn't state it very well.
@sjaindl, can you reopen this Pull Request now (before! you rebase).

**Comment by sjaindl** [comment]
@robertpainsi I can't see an option to reopen the PR, neither before nor after rebasing (at least it seems that I don't have the permission to do so, if possible). Is it possible for you?

**Comment by robertpainsi** [comment]
@sjaindl
Thank you for the info!

> Is it possible for you?

After a force push, no :smile_cat: 

@thmq 
Most developers (developers without write access) can't reopen their Pull Request once they are closed. So they have to create new ones. All previous code review comments, discussions and other information are located in older Pull Requests. So everything is scattered in multiple Pull Requests. :crying_cat_face:
Maybe adapt this?

**Pull request closed without merging**